<?xml version="1.0"?>
<!-- 
    Bringup launch file for SLAM nodes
 -->
<launch>
    <!-- Global Arguments: -->
    <arg name="log_level"               default="2"/>
    <arg name="reset"                   default="True"/>
    <arg name="launch_slam_node"        default="true"/>
    <arg name="launch_slam_ekf"         default="true"/>
    <arg name="launch_transformer"      default="true"/>
    <arg name="load_params"             default="false"/>
    <arg name="param_file"              default="sim_cfg.yaml"/>

    <arg name="namespace"               default="/vpr_nodes"/>
    
    <group if="$(arg load_params)">
        <!-- Load in params; must be done outside of a namespace -->
        <rosparam command="load" file="$(find aarapsi_robot_pack)/cfg/$(arg param_file)"/> 
    </group>

    <group ns="$(arg namespace)">
        <!-- SLAM Localization Nodes: -->
        <group if="$(arg launch_slam_node)">
            <include file="$(find aarapsi_robot_pack)/launch/hdl/hdl_localization.launch">
                <arg name="param_file"  value="$(arg param_file)"/>
            </include>
        </group>

        <!-- SLAM EKF Node: -->
        <group if="$(arg launch_slam_ekf)">
            <include file="$(find aarapsi_robot_pack)/launch/hdl/hdl_ekf.launch"/>
        </group>

        <!-- Frame Transformer Node: -->
        <group if="$(arg launch_transformer)">
            <node pkg="aarapsi_robot_pack" type="frame_transformer.py" name="frame_transformer" output="screen"
                args="-V $(arg log_level) -R $(arg reset) -ID 0"/>
        </group>

    </group>

</launch>